packages/synapse-react-client/src/components/QueryWrapper/useQueryWrapperPaginationControls.ts

```diff
@@ -0,0 +1,111 @@
+import { ImmutableTableQueryResult } from '../useImmutableTableQuery'
+import { UseQueryWrapperDataReturn } from './useQueryWrapperData'
+
+type PaginatedControls = {
+  /** Navigates to a particular page, where the first page has value 1 */
+  goToPage: (pageNum: number) => void
+  /** The current page number, where page 1 is the first page. */
+  currentPage: number
+  /** Updates the page size */
+  setPageSize: (pageSize: number) => void
+  /** The current page size. */
+  pageSize: number
+}
+
+type InfiniteQueryPaginationControls = {
+  /** Returns true when loading a new page of query results */
+  isLoadingNewPage: boolean
+  /** Whether the query result bundle has a next page */
+  hasNextPage?: boolean
+  /** Invoke this method to fetch and append the next page of rows to the data  */
+  appendNextPageToResults: () => Promise<void>
+  /** Invoke to fetch and update the data with the next page of query results */
+  goToNextPage: () => Promise<void>
+  /** Whether the query result bundle has a previous page */
+  hasPreviousPage: boolean
+  /** Invoke to fetch and update the data with the previous page of query results */
+  goToPreviousPage: () => Promise<void>
```

Comment:
InfiniteQueryWrapper provided these custom functions to manage pagination, so we need to continue to provide them to limit the scope of this change.

The hook in this file will provide these controls when we have an infinite data query.